dan_layer/core/src/workers/hotstuff_waiter.rs

```diff
@@ -390,12 +395,15 @@ where
             .await?;
 
         let mut votes_to_send = vec![];
-        let vn_mmr = self.epoch_manager.get_validator_node_mmr(node.epoch()).await?;
         let vns = self.epoch_manager.get_validator_nodes_per_epoch(node.epoch()).await?;
         let vn_mmr_leaf_index = vns
             .iter()
-            .position(|vn| vn.public_key == *self.node_identity.public_key())
+            .position(|vn| vn.public_key == self.public_key)
             .expect("The VN is not registered");
+
+        let node_shard_id = self.get_shard_id(node.epoch(), self.public_key.clone()).await?;
```

Comment:
I wasn't sure how often this was going to be called, but every proposal will be devastatingly slow

Reply:
yeah we need a local cache for this

Reply:
Changed to use the shard key in the local db